qcore/utils.py

```diff
@@ -6,8 +6,149 @@
 from shutil import rmtree
 import os
 import imp
+import yaml
+from collections import OrderedDict
+from collections import Mapping
 
-def setup_dir(directory, empty = False):
+
+class DotDictify(dict):
```

Comment:
is there any point to make a whole bunch of complexity just to do d.k instead of d['k']?? There needs to be a legitimate reason to do this

Reply:
Fair point. And that was discussed a while ago. For backwards / forwards compatibility reasons it was decided to support both in the first instance.